### PR TITLE
I. #852 add TG logging to workbench settings.

### LINKED ISSUE #852: Expose a setting for Taskgraph's logging level
We offer this in Qt; the ability to set the Taskgraph loggingLevel separately from all other loggers, because DEBUG logging from Taskgraph is so noisy.

### PR DESCRIPTION
**Pull request by dcdenu4**
Adds taskgraph logging threshold to the Workbench settings modal.  Exposes taskgraph logging through the invest CLI via ``--taskgraph-log-level``.

Added tests similar to ``Logging threshold`` setting.

## Description
Fixes #852

## Checklist
- [x] Updated HISTORY.rst (if these changes are user-facing)
- [ ] Updated the user's guide (if needed)
- [x] Tested the affected models' UIs (if relevant)

### CODE REVIEW

**Comment by dcdenu4** [comment]
Thanks @davemfish , I think I'm going to punt on those issues you mentioned for now. I made issues for them in #950 and #951 . The issue of TG logging being overridden by InVEST logging is seemingly a QT UI thing too, so it's a larger logging issue then just this PR. The issue of differentiating InVEST logging between the UI window and file handler feels like it warrants a different PR.

Otherwise I think I addressed your comments!

File: src/natcap/invest/cli.py [hunk 1]
```diff
@@ -354,6 +361,12 @@ def main(user_args=None):
     root_logger.addHandler(handler)
     LOGGER.info('Setting handler log level to %s', log_level)
 
+    # Set the log level for taskgraph.
+    tg_logging_map = {'DEBUG': 10, 'INFO': 20, 'WARNING': 30, 'ERROR': 40}
+    taskgraph_log_level = tg_logging_map[args.taskgraph_log_level.upper()]
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply src/natcap/invest/cli.py]: Nice! Fixed in https://github.com/natcap/invest/pull/946/commits/9978f0fedafca73e92f93ceb6fcf4f0567cf9afa

File: src/natcap/invest/cli.py [hunk 2]
```diff
@@ -354,6 +361,12 @@ def main(user_args=None):
...
+    logging.getLogger('taskgraph').setLevel(taskgraph_log_level)
+    LOGGER.info('Setting taskgraph log level to %s', taskgraph_log_level)
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply src/natcap/invest/cli.py]: Changed in https://github.com/natcap/invest/pull/946/commits/2e548b7044704c2653302fd9342db6eb22e59b1e

File: workbench/src/renderer/components/SettingsModal/SettingsStorage.js [hunk 3]
```diff
@@ -10,12 +10,14 @@ const investSettingsStore = localforage.createInstance({
...
  * @returns {object} to destructure into:
  *     {String} nWorkers - TaskGraph number of workers
+ *     {String} taskgraphLoggingLevel - InVEST taskgraph logging level
  *     {String} loggingLevel - InVEST model logging level
  *     {String} sampleDataDir - default location for sample datastack downloads
  */
 export function getDefaultSettings() {
   const defaultSettings = {
     nWorkers: '-1',
+    taskgraphLoggingLevel: 'ERROR',
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply workbench/src/renderer/components/SettingsModal/SettingsStorage.js]: That sounds right to me, I think QT UI was using `ERROR` but `INFO` seems like a more useful default. Changed in https://github.com/natcap/invest/pull/946/commits/fb6ef0a6ae1640a55b3ffc10a896965b3af4819c

File: workbench/tests/renderer/app.test.js [hunk 4]
```diff
@@ -405,13 +407,16 @@ describe('InVEST global settings: dialog interactions', () => {
 
     userEvent.click(await findByRole('button', { name: 'settings' }));
     const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
-    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: true });
+    const tgLoggingInput = getByLabelText(tgLoggingLabelText, { exact: true });
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply workbench/tests/renderer/app.test.js]: Good call. Changed in https://github.com/natcap/invest/pull/946/commits/79b832e38d97b46e222b25958c70d20a487b827b

File: workbench/tests/renderer/app.test.js [hunk 5]
```diff
@@ -450,13 +459,15 @@ describe('InVEST global settings: dialog interactions', () => {
...
```
[the remark this answers is not in the document]
Comment (dcdenu4) [reply workbench/tests/renderer/app.test.js]: Changed in https://github.com/natcap/invest/pull/946/commits/79b832e38d97b46e222b25958c70d20a487b827b

**Review by davemfish — APPROVED** [review]

**Pull request merged**